src/AutoRest.CSharp/Common/Output/Builders/MultipartSerializationMethodsBuilder.cs

```diff
@@ -0,0 +1,265 @@
+        };
+        private static MethodBodyStatement SerializeArray(VariableReference mulitpartContent, MultipartArraySerialization serialization, ValueExpression value, StringExpression name)
+        {
+            return new[]
+            {
+                new ForeachStatement(serialization.Type.ElementType, "item", value, false, out var item)
+                {
+                    SerializationExression(mulitpartContent, serialization.ValueSerialization, item, name)
+                }
+            };
+        }
+        private static MethodBodyStatement SerializeDictionary(VariableReference mulitpartContent, MultipartDictionarySerialization serialization, ValueExpression value, StringExpression name)
+        {
+            var dictionaryExpression = new DictionaryExpression(serialization.Type.Arguments[0], serialization.Type.Arguments[1], value);
+            return new[]
+            {
+                new ForeachStatement("item", dictionaryExpression, out KeyValuePairExpression keyValuePair)
+                {
+                    SerializationExression(mulitpartContent, serialization.ValueSerialization, keyValuePair.Value, new StringExpression(keyValuePair.Key))
+                }
+            };
+        }
+
+        private static MethodBodyStatement SerializeValue(VariableReference mulitpartContent, MultipartValueSerialization valueSerialization, ValueExpression valueExpression, StringExpression name)
+        {
+            ValueExpression? contentExpression = null;
+            ValueExpression? fileNameExpression = null;
+            ValueExpression? contentTypeExpression = null;
+            if (valueSerialization.Type.IsFrameworkType && valueSerialization.Type.FrameworkType == typeof(Stream))
+            {
+                contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+                if (valueSerialization.FileName != null)
+                {
+                    fileNameExpression = Literal(valueSerialization.FileName);
+                }
+                else
+                {
+                    fileNameExpression = name.Add(Literal(".wav"));
+                }
+                if (valueSerialization.ContentType != "application/json")
+                {
+                    contentTypeExpression = Literal(valueSerialization.ContentType);
+                }
+            }
+            if (valueSerialization.Type.IsFrameworkType && valueSerialization.Type.FrameworkType == typeof(BinaryData))
+            {
+                contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+                if (valueSerialization.FileName != null)
+                {
+                    fileNameExpression = Literal(valueSerialization.FileName);
+                } else
+                {
+                    fileNameExpression = name.Add(Literal(".wav"));
+                }
+                if (valueSerialization.ContentType != "application/json")
```

Comment:
we have quite a few magic strings.
For `application/json`, I think we should define it somewhere.
For `.wav` is this correct? This looks so weird.